File: smb:/192.168.99.130/staging/Release/System/Data/Models/Application.js
var store = require("../Storage");
exports.Table = undefined;
var Application = (function () {
    function Application() {
        this.uid = "";
        this.appsig = "";
        this.name = "";
        this.urlName = "";
    }
    Application.table = function () {
        if (!exports.Table) {
            exports.Table = store.DefineTable("Application", Application, { id: ["uid"] });
        }
        return exports.Table;
    };
    Application.Strip = function (App) {
        return {
            uid: App.uid
        };
    };
    return Application;
})();
exports.Application = Application;

File: System/DB/Models/Application.lua
require 'Coat.Persistent'
require 'Coat.Persistent'.trace = print
local uuid = require 'uuid'

persistent "Application"
has_p.uid = { is = 'rw', isa = 'string' }
has_p.appsig = { is = 'rw', isa = 'string' }
has_p.name = { is = 'rw', isa = 'string' }
has_p.urlName = { is = 'rw', isa = 'string' }


sql_create = [[
    CREATE TABLE Application (
        id INTEGER PRIMARY KEY AUTOINCREMENT,
        uid CHAR(64),
        appsig CHAR(32),
        name CHAR(255),
        urlName CHAR(255)
    )
]]

local exports = {}

function exports.init(dbpath)
    local conn = Application.establish_connection('sqlite3', dbpath)
    conn:execute(Application.sql_create)

    return Application
end

return exports

